feature/BRUI-430_Snake_in_BRUI_Ladebildschirm_view3_view4_einbauen

holzmare opened the pull request:
Pull request is sadly pretty fragmented with very granular commits in the beginning, because it is based on previous tickets that had not been merged to develop yet
Changes:
- made snake a package in separate ticket, based this branch of it
- implemented EndScreen in separate ticket, merged here
- added babel to project
- added esbuild to bundle the project
- new command "npm run package" to babel and bundle
- npm package is based on resulting /lib/main.js file
- implemented destroy() method logic
- added functionality to EndScreen buttons
- restructured field
- moved most styling to styling methods for readability
- added .nvmrc file to ensure correct node version with nvm
- added SnakeGame.status to track game status
- fixed EndScreen scaling issues
- fixed Field.clear() not resetting background color correctly

holzmare committed:
feature/DBI-1143_DBISnake_Endscreen_verbessern
- Endscreen implementiert
- Einbindung Endscreen WIP

holzmare committed:
feature/DBI-1143_DBISnake_Endscreen_verbessern
- gave Endscreen Buttons functionality
- implemented destroy methods

holzmare committed:
feature/DBI-1143_DBISnake_Endscreen_verbessern
- fixed bug where snake could move at double speed when replay got
  clicked before start loop ended

holzmare committed:
feature/DBI-1143_DBISnake_Endscreen_verbessern
- moved around code to prevent problems with the snake crashing when
  restarting the game

holzmare committed:
feature/DBI-1143_DBISnake_Endscreen_verbessern
- set style via .style. instead of setAttribute in a few places
- moved handling of config object to separate method

holzmare committed:
feature/BRUI-430_Snake_in_BRUI_Ladebildschirm_view3_view4_einbauen
- added babel
- based npm package on babel output

holzmare committed:
feature/BRUI-430_Snake_in_BRUI_Ladebildschirm_view3_view4_einbauen
- restructured Field
- changed container to have grid display

holzmare committed:
feature/BRUI-430_Snake_in_BRUI_Ladebildschirm_view3_view4_einbauen
- added babel installation to package.json
- rebuilt field with previuous changes

holzmare committed:
feature/BRUI-430_Snake_in_BRUI_Ladebildschirm_view3_view4_einbauen
- changed babel config
- added .nvmrc file

holzmare committed:
feature/BRUI-430_Snake_in_BRUI_Ladebildschirm_view3_view4_einbauen
- new babel config for testing

holzmare committed:
Update package.json

feature/BRUI-430_Snake_in_BRUI_Ladebildschirm_view3_view4_einbauen/package.json
- fixed error in package.json

holzmare committed:
feature/BRUI-430_Snake_in_BRUI_Ladebildschirm_view3_view4_einbauen
- bundled with esbuild

holzmare committed:
Merge branch 'feature/BRUI-430_Snake_in_BRUI_Ladebildschirm_view3_view4_einbauen' of github.com:holzmare/DBIsnake into feature/BRUI-430_Snake_in_BRUI_Ladebildschirm_view3_view4_einbauen

holzmare committed:
feature/BRUI-430_Snake_in_BRUI_Ladebildschirm_view3_view4_einbauen
- fixed package command in package.json
- added /bundle to .gitignore
- fixed scaling issues with EndScreen
- mo0ved EndScreen buttons styling to own function
- fixed fied.clear() function
- added SnakeGame.state to track game status

holzmare committed:
feature/BRUI-430_Snake_in_BRUI_Ladebildschirm_view3_view4_einbauen
- refactored status codes to make more sense

MaxGeldner reviewed (approved):
- Field construuctor struktuurieren/kommentieren
- Status am Ende der Aktionen setzen und eventuuell duurch Strings auustauuschen (optional)

holzmare committed:
feature/BRUI-430_Snake_in_BRUI_Ladebildschirm_view3_view4_einbauen

- changed when some of the status are set

holzmare committed:
feature/BRUI-430_Snake_in_BRUI_Ladebildschirm_view3_view4_einbauen

- rearranged Button creation code
- commented blocks in constructor to improve radability

holzmare committed:
Merge branch 'main' into feature/BRUI-430_Snake_in_BRUI_Ladebildschirm_view3_view4_einbauen
